### PR TITLE
DT-1377 Refactor to allow a filter to be applied to documents

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 1]
```diff
@@ -507,7 +446,7 @@ public CustodialStatus getCustodialStatusBySentenceId(
 
     private ResponseEntity<AccessLimitation> accessLimitationResponseEntityOf(final OffenderDetail offender) {
 
-        final var accessLimitation = userService.accessLimitationOf(currentUserSupplier.username().get(), offender);
+        final var accessLimitation = userService.accessLimitationOf(currentUserSupplier.username().orElseThrow(), offender);
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:449]: Check there is a test that covers this scenario

File: src/main/java/uk/gov/justice/digital/delius/service/DocumentService.java [hunk 2]
```diff
@@ -220,6 +224,62 @@ public OffenderDocuments offenderDocumentsFor(Long offenderId) {
                 .build();
     }
 
+    private List<PersonalCircumstanceDocument> personalCircumstanceDocumentsFor(Long offenderId) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/DocumentService.java:227]: missed filter

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java (hunk 1 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:449]: Both get() and orElseThrow will throw same exception NoSuchElementException

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/DocumentResource.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+@AllArgsConstructor
+@PreAuthorize("hasRole('ROLE_COMMUNITY')")
+public class DocumentResource {
+
+    private final OffenderService offenderService;
+    private final AlfrescoService alfrescoService;
+    private final DocumentService documentService;
+
+
+    @ApiOperation(value = "Returns all document's meta data for an offender by NOMS number", tags = "Documents")
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/DocumentResource.java:43]: moved from OffendersResource

File: src/main/java/uk/gov/justice/digital/delius/service/DocumentService.java [hunk 4]
```diff
@@ -220,6 +224,62 @@ public OffenderDocuments offenderDocumentsFor(Long offenderId) {
                 .build();
     }
 
+    private List<PersonalCircumstanceDocument> personalCircumstanceDocumentsFor(Long offenderId, DocumentFilter filter) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/DocumentService.java:227]: TODO use the filter

File: src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java [hunk 5]
```diff
@@ -56,7 +52,7 @@ void willCheckUserAccessIsNotRestrictedOrExcluded(){
         final var offender = theOffender();
         given(offenderService.getOffenderByCrn(SOME_CRN_NUMBER)).willReturn(offender);
         given(currentUserSupplier.username()).willReturn(Optional.of("BOB"));
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java:54]: IntelliJ warning fixes

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**